### DISCUSSION TITLE
Call history / missed calls

### DISCUSSION THREAD

**Discussion by WebworkrNet** [FossifyOrg/General-Discussion#533] [Q&A]
Samsung Galaxy A13 5G (SM-A136B)
Android 14, UI-Version 6

Samsung's default phone app cannot be uninstalled or deactivated. I have already set the Fossify dialer to standard. However, the call list remains empty and missed calls are still displayed in the Samsung app. Is there anything I can improve on the settings or is it due to technical circumstances? 

Thank you very much

![Screenshot_20240308_105814_Permission controller.jpg](https://github.com/FossifyOrg/Phone/assets/43509007/868cdfd4-ecff-4806-9e7f-2901123d2432)

**Comment by Aga-C** [comment]
@naveensingh Maybe you will be able to help.

**Comment by naveensingh** [comment]
@WebworkrNet Can you check if **Call logs** permission is granted to Fossify Phone? Go to Settings -> Apps -> Fossify Phone -> Permissions -> Call logs -> Allow. 

Usually, all required permissions are granted automatically when the app becomes the default phone app unless the user manually revokes any permissions. I just verified this again on a Samsung phone running Android 14 and One UI 6.0.

**Reply by WebworkrNet** [reply]
In my opinion, all the necessary permissions have been granted.

![Screenshot_20240308_215822_Permission controller.jpg](https://github.com/FossifyOrg/Phone/assets/43509007/03d8ab6d-0da3-44aa-b954-9f77dbd8b53d)

I'm also having trouble answering a call. Neither dialer offers an option to accept the call. It rings via Fossify, but I have to actively open the app to accept the call - neither a full view in the foreground nor a pop-up. The call is displayed in the status bar, I seem to remember.

**Reply by naveensingh** [reply]
Some Samsung devices have some special voodoo that break some functionalities. 

Please try granting all permissions required by phone-ish apps from the system:

<img src="https://github.com/FossifyOrg/Phone/assets/36371707/63aae423-80f8-4817-aa1e-16f8235052fe" width=264 /> <img src="https://github.com/FossifyOrg/Phone/assets/36371707/9fa50b6c-e9b4-4710-9b77-8bc615f1caa9" width=264 />

If it still doesn't work, try reinstalling Fossify Phone.

**Reply by WebworkrNet** [reply]
Do you mean permission manager?

![Screenshot_20240308_233740_Permission controller.jpg](https://github.com/FossifyOrg/Phone/assets/43509007/0740302c-81da-4fa9-85b5-0499e5deb254)

By the way, if you notice a "suspicious" authorization, I would be grateful for a tip. I try to regularly remove unnecessary and privacy-critical permissions.

**Reply by naveensingh** [reply]
> Do you mean permission manager?

I meant Settings -> Apps -> Show system (in the menu) -> Search "phone" and allow all permissions for all apps that are in the search results. Refer to my screenshots, it's from a Samsung phone.

> if you notice a "suspicious" authorization

Looks good to me!

**Reply by yusufsunetci** [reply]
Open the application. Settings - Calls - Always display incoming calls in full screen.

When the call came, I couldn't accept it without opening the app. When I did that, it work.

**Reply by WebworkrNet** [reply]
> Open the application. Settings - Calls - Always display incoming calls in full screen.

> When the call came, I couldn't accept it without opening the app. When I did that, it work.

I have made the suggested setting and will continue to monitor the app behavior.

**Reply by WebworkrNet** [reply]
> Open the application. Settings - Calls - Always display incoming calls in full screen.

> When the call came, I couldn't accept it without opening the app. When I did that, it work.

There has been an improvement in that it is now possible to accept calls.

However, I feel that it takes too long for the synchronization with the contacts to take place. Initially, only the number is displayed. I have integrated many address books and contacts. The Samsung standard dialer can also handle this. In my memory, incoming calls from contacts are always displayed as such immediately. This is probably another problem.

Another shortcoming: Push notifications about missed calls are still sent by the standard dialer.

**Reply by WebworkrNet** [reply]
Yesterday I noticed that while Fossify-Phone was open in the foreground, it was not possible to answer an incoming call. 
This was probably due to the fact that Fossify was still trying to establish a connection to another subscriber in the background, which I only recognized later from the push message. I then made the call via the Samsung dialer instead.

@naveensingh Do you still think that a reinstallation would be helpful after the latest reports and confirmation by yusufsunetci ?

**Reply by WebworkrNet** [reply]
@yusufsunetci What kind of device are you using? 

Thanks

**Reply by yusufsunetci** [reply]
Xiaomi Note 8 Pro

**Comment by WebworkrNet** [comment]
Some permissions are restricted only with the standard dialer. I find it difficult to imagine that these settings have any effect.

![Screenshot_20240310_213257_Permission controller](https://github.com/FossifyOrg/Phone/assets/43509007/b58f7753-5ae9-47cf-9308-3aabfdabf725)


All other apps have all the necessary permissions or do not require any.

**Comment by naveensingh** [comment]
Do you see "Could not access the call history" on the **Call history** tab? If yes, what happens when you tap "Request access"?

It doesn't seem to be be an issue with the app. The latest Android 14 update also broke media permission intent on my samsung phone (the settings app just crashes now).

**Comment by WebworkrNet** [comment]
Weird acting:
After I took the following screenshot, the call list filled up unexpectedly and without any further action. You can see from the screenshot that the list was previously empty and there was no problem with authorization. (In this case I did not switch the language to English on the phone beforehand, I hope it is still clear what is meant).

![Screenshot_20240311_105736_Phone](https://github.com/FossifyOrg/Phone/assets/43509007/7916b3dc-8482-41df-8fce-67bfb3e98630)

I'll see how the app behaves with new outgoing and incoming calls from now on.

**Reply by naveensingh** [reply]
Good to know that it's working now. Let us know if you learn anything about it, whether it's an app issue or system issue.

**Reply by WebworkrNet** [reply]
I'll give you feedback on what my observations reveal. If I notice anything and/or new insights emerge, I will share it.

**Reply by WebworkrNet** [reply]
The content of the call list varies: At first it is empty, then presumably after opening the app several times or bringing it to the foreground, it is filled.

For today's incoming call, neither dialer opened, even though it rang. I actively opened the Fossify dialer, but there was no option to answer the call. Then I opened the Samsung dialer, whereupon, as expected, I was initially asked whether the default app should be changed. I skipped this step. I can vaguely remember the following point. Either it was not possible to accept the call or the other participant had hung up in the meantime. I then called back based on the missed call push notification. This notification is from Fossify Phone according to notification history.

As already mentioned at the beginning, is the only option left to uninstall and reinstall?

**Reply by WebworkrNet** [reply]
The behaviour can really be described as strange. Today I received a call via the Fossify app and was able to answer it directly without any problems.

(The only unusual thing was that the sound of my TV continued to stream live in the background, but that's another "construction site").

**Comment by WebworkrNet** [comment]
Recently, the call acceptance did not work again and the call list was also empty. I seem to remember that this also applied to the "Contacts" and "Favorites" tabs. In short: sometimes it works, sometimes it doesn't.

**Comment by WebworkrNet** [comment]
**Contacts load with a long delay or not at all** 
https://github.com/FossifyOrg/General-Discussion/issues/220
